### PR TITLE
[JAV-560] fix null header process logic on consumer side

### PR DESCRIPTION
**Pull request by yhs0092**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://https://servicecomb.atlassian.net/projects/JAV/issues) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[JAV-XXX] Fixes bug in ApproximateQuantiles`, where you replace `JAV-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

see details in [https://servicecomb.atlassian.net/browse/JAV-560]

Check null arguments in invocation and do not copy them into requests to avoid NullPointerException in Netty.
My test demo is here: [https://github.com/yhs0092/javachassis-demo-header-transport]

### CODE REVIEW

File: providers/provider-springmvc/src/main/java/io/servicecomb/provider/springmvc/reference/RestTemplateCopyHeaderFilter.java [hunk 1]
```diff
@@ -40,6 +40,10 @@ public void beforeSendRequest(Invocation invocation, HttpServletRequestEx reques
 
     httpHeaders.forEach((key, values) -> {
       for (String value : values) {
+        // null args should not be set to requestEx to avoid NullPointerException in Netty.
+        if (null == value) {
+          continue;
```
Comment (WillemJiang) [inline providers/provider-springmvc/src/main/java/io/servicecomb/provider/springmvc/reference/RestTemplateCopyHeaderFilter.java]: Please do some debug log to help the user tracing the setting issue.

File: common/common-rest/src/main/java/io/servicecomb/common/rest/codec/param/HeaderProcessorCreator.java [hunk 2]
```diff
@@ -56,6 +56,10 @@ public Object getValue(HttpServletRequest request) throws Exception {
 
     @Override
     public void setValue(RestClientRequest clientRequest, Object arg) throws Exception {
+      if (null == arg) {
+        // null header should not be set to clientRequest to avoid NullPointerException in Netty.
```
Comment (WillemJiang) [inline common/common-rest/src/main/java/io/servicecomb/common/rest/codec/param/HeaderProcessorCreator.java]: We need to do some debug log here.

Comment (yhs0092) [reply common/common-rest/src/main/java/io/servicecomb/common/rest/codec/param/HeaderProcessorCreator.java]: debug log added, please review it.

File: providers/provider-springmvc/src/main/java/io/servicecomb/provider/springmvc/reference/RestTemplateCopyHeaderFilter.java (hunk 1 above)
Comment (yhs0092) [reply providers/provider-springmvc/src/main/java/io/servicecomb/provider/springmvc/reference/RestTemplateCopyHeaderFilter.java]: debug log added, please review it.

**Pull request merged**